src/AppInstallerRepositoryCore/Public/winget/RepositorySearch.h

```diff
@@ -353,6 +352,22 @@ namespace AppInstaller::Repository
         virtual const void* CastTo(IPackageType type) const = 0;
     };
 
+    // Contains information about a package from a search result.
+    struct IPackageSearchResult
+    {
+        virtual ~IPackageSearchResult() = default;
+
+        // Gets a property of this package result.
+        virtual Utility::LocIndString GetProperty(PackageProperty property) const = 0;
+
+        // Gets the installed package information.
+        virtual std::shared_ptr<IPackage> GetInstalled() const = 0;
+
+        // Gets all of the available packages for this result.
+        // There will be at most one package per source in this list.
+        virtual std::vector<std::shared_ptr<IPackage>> GetAvailable() const = 0;

```

Comment:
So just wanted to confirm, in the past for correlation, when we find 1 matching remote available, we stopped matching for more. Are we going to do matching of multiple remote packages?

Reply:
Yes, the goal is that the object returned by a search contains the correlated installed and available items.  I don't want to do more searching than we already do though, so I'm not going to guarantee that it actually contains the complete set.

For instance, when we find some installed item, if it is available from multiple sources (and correlates properly) we will have a single result with a package entry for all of them.

If we find an available item that correlates to the same installed item as another available source item, then we will add it to the list.  But those same two available items will not be correlated into a single result (at this time) if there is no installed item.  And I definitely wouldn't expect to search for the ID of a package in one source and have it return a result with the available package that correlates in every other source (although it is possible that this could happen in the future).